ext/fg/js/frontend-initialize.js

```diff
@@ -29,9 +31,32 @@ async function main() {
     const {id, depth=0, parentFrameId, ignoreNodes, url, proxy=false} = data;
 
     let popup;
-    if (proxy) {
+    if (!proxy && (window !== window.parent)) {
+        let rootPopupInformationResolve;
+        const rootPopupInformationPromise = new Promise((resolve) => (rootPopupInformationResolve = resolve));
+
+        const runtimeMessageCallback = ({action, params}, sender, callback) => {
```

Comment:
We have this pattern in a few places now; we may want to abstract it into something that auto-removes the listener. Something like `yomichan.addListenerTemporary(chrome.runtime.onMessage, callback)`. Not necessary to do it now if you think it will make the PR more complex.

Reply:
I'll see if it can be easily done to the places this PR uses, and changed to other places later.

Reply:
https://github.com/FooSoft/yomichan/pull/417/commits/9fdcedb4001a5e86cc7e648c61bd9961a202fa54 implements an abstraction for the pattern that was used here and also in `Popup._listenForDisplayPrepareCompleted`, but that wasn't converted to use the new abstraction yet.